jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/gcprules/vminstance/VMInstanceServiceAccountCloudAPIAccess.java

```diff
@@ -0,0 +1,128 @@
+        }
+
+        if (!StringUtils.isNullOrEmpty(vmEsURL)) {
+            vmEsURL = vmEsURL + "/gcp_vminstance/_search";
+        }
+        logger.debug("========vmEsURL URL after concatenation param {}  =========", vmEsURL);
+
+        boolean doesVMServiceAccountHaveFullCloudApiAccess = false;
+
+        MDC.put("executionId", ruleParam.get("executionId"));
+        MDC.put("ruleId", ruleParam.get(PacmanSdkConstants.RULE_ID));
+
+        if (!StringUtils.isNullOrEmpty(resourceId)) {
+
+            Map<String, Object> mustFilter = new HashMap<>();
+            mustFilter.put(PacmanUtils.convertAttributetoKeyword(PacmanRuleConstants.RESOURCE_ID), resourceId);
+            mustFilter.put(PacmanRuleConstants.LATEST, true);
+
+            try {
+                doesVMServiceAccountHaveFullCloudApiAccess = verifyVmServiceAccountHaveFullCloudApiAccess(vmEsURL, mustFilter);
+                if (doesVMServiceAccountHaveFullCloudApiAccess) {
+                    List<LinkedHashMap<String, Object>> issueList = new ArrayList<>();
+                    LinkedHashMap<String, Object> issue = new LinkedHashMap<>();
+
+                    annotation = Annotation.buildAnnotation(ruleParam, Annotation.Type.ISSUE);
+                    annotation.put(PacmanSdkConstants.DESCRIPTION, "Enforce the principle of least privilege to prevent privilege escalation, ensure that compute engine instances are not configured to use default service account with the cloud api access scope set to \"Allow full acess to all cloud Apis");
+                    annotation.put(PacmanRuleConstants.SEVERITY, severity);
+                    annotation.put(PacmanRuleConstants.CATEGORY, category);
+
+                    issue.put(PacmanRuleConstants.VIOLATION_REASON, "VM instance service account has cloud api access");
+                    issueList.add(issue);
+                    annotation.put("issueDetails", issueList.toString());
+                    logger.debug("========VMInstanceServiceAccountCloudAPIAccess ended with an annotation {} : =========", annotation);
+                    return new RuleResult(PacmanSdkConstants.STATUS_FAILURE, PacmanRuleConstants.FAILURE_MESSAGE, annotation);
+                }
+
+            } catch (Exception exception) {
+                throw new RuleExecutionFailedExeption(exception.getMessage());
+            }
+        }
+
+        logger.debug("========VMInstanceServiceAccountCloudAPIAccess ended=========");
+        return new RuleResult(PacmanSdkConstants.STATUS_SUCCESS, PacmanRuleConstants.SUCCESS_MESSAGE);
+
+    }
+
+    private boolean verifyVmServiceAccountHaveFullCloudApiAccess(String vmEsURL, Map<String, Object> mustFilter) throws Exception {
+        logger.debug("========verifyVmServiceAccountHaveFullCloudApiAccess started=========");
+        JsonArray hitsJsonArray = GCPUtils.getHitsArrayFromEs(vmEsURL, mustFilter);
+        boolean validationResult = false;
```

Comment:
Rule looks fine,can we also skip this validation for VM created by GKE.

Reply:
added validation for gke